### ISSUE TITLE
Objects do not have a prototype

### ISSUE THREAD

**Issue by mattapperson** [TiForward/HAL#5]
Native objects (for example the Widget in the Widget example) do not have a prototype. Not a deal breaker, but we go back to the whole "jot really JS objects" bit

**Comment by yuchi** [comment]
This is way more fun than it looks!

/cc @omorandi Let’s solve it once for all.

**Comment by yuchi** [comment]
Here’s the (old) related Hyperloop appcelerator/hyperloop#92
(didn’t even had to search for it, I remembered the issue number…)

**Comment by mattapperson** [comment]
@yuchi that just means you need more of a life outside of programing :P

**Comment by yuchi** [comment]
I have one! I’m into this ‘developing’ thing also!

**Comment by mattapperson** [comment]
Hey im not complaining... if anything you need to get some more coding done so we can kick this Ti.Next thing out the door :P

**Comment by omorandi** [comment]
Keeping @yuchi's life aside, I think this can be a non-issue. The fact that in the widget example the widget object doesn't have a prototype only means that it wasn't programmed to have one.
The issue @yuchi is referring in hyperloop was in a completely different context of HL target code being generated in a (wrong) way where the prototype chain was messed up.
In other words, it's not a HAL problem if injected objects don't have a prototype. It's rather a HAL programmer's responsibility to make it so.

**Comment by mattapperson** [comment]
@omorandi its not a high priority issues... but I do think it is an issue. HAL and HL should be able to support prototypes. This is an expected part of the JS language.

**Comment by omorandi** [comment]
I agree on this, but I only wanted to point out that it's not strictly a HAL issue.

**Comment by mattapperson** [comment]
thats fair

**Comment by yuchi** [comment]
@omorandi couldn't have said it better.

The biggest challenge is not having a working prototype, but having a sane prototype chain.
